de.metas.business/src/main/java-legacy/org/compiere/model/MPayment.java

```diff
@@ -2271,66 +2235,9 @@ private boolean allocatePaySelection()
 	{
 		MAllocationHdr alloc = new MAllocationHdr(getCtx(), false,
 				getDateTrx(), getC_Currency_ID(),
-				Msg.translate(getCtx(), "C_Payment_ID") + ": " + getDocumentNo() + " [n]", get_TrxName());
+				Services.get(IMsgBL.class).translate(getCtx(), "C_Payment_ID") + ": " + getDocumentNo() + " [n]", get_TrxName());
 		alloc.setAD_Org_ID(getAD_Org_ID());
 
-		String sql = "SELECT psc.C_BPartner_ID, psl.C_Invoice_ID, psl.IsSOTrx, "	// 1..3
```

Comment:
u just removed the code which is creating the C_AllocationLines. So, here we will end up with an empty allocation which will lead to error.

IMHO, "allocatePaySelection" method does not make sense anymore. Pls delete it.

Reply:
Done. Thanks!